### ISSUE TITLE
Next.js + Express example can't be deployed to Now

### ISSUE THREAD

**Issue by tiotdev** [zeit/now-examples#196]
# Examples bug report

## Example name
Express example (https://github.com/zeit/next.js/tree/canary/examples/custom-server-express) 

## Describe the bug
Can't deploy to now

## To Reproduce
The "deploy to now button" takes me to a site where I can enter an API token and press deploy. It says "Initialising" - and then nothing happens and it goes back to where I should enter an API token.

So I tried it manually:
- Downloading the example, `npm install`, adding the now.json file
`{
  "version": 2,
  "name": "nodejs-express",
  "builds": [{ "src": "server.js", "use": "@now/node-server" }]
}` (source: https://zeit.co/examples/nodejs-express/)
returns a directory listing with the "server.js" file.

Changing the filename from "server.js" to "index.js" and changing this as well in package.json and now.json, running `npm run build` and then `now` generates a HTTP ERROR 500.

## Expected behaviour
I should be able to deploy the example

**Comment by timneutkens** [comment]
Tranfered the issue to now-examples.

**Comment by rauchg** [comment]
This however related to the "Next.js + Express" example. Updating the title

**Comment by TJBlackman** [comment]
So happy this is a real issue. I've been at this for a couple days and here's what I've learned. 
Pure Express apps can be launched successfully. 
It's only when I try to incorporate the NextJs from the demo that I get the 500 errors. 

If I get the logs for the deployment, the error the 500 error. When I run logs on the deployment, I see this message: `Could not find a valid build in the '/var/task/user/.next' directory! Try building your app with 'next build' before starting the server.`

I tried adding the now.json file as OP did, but I still get the error. :(

**Comment by rauchg** [comment]
@TJBlackman we'll definitely take a look at this one, but the `@now/next` builder (which makes Next.js *truly* serverless) is the priority at the moment

It's possible that maybe we're not copying over `.` directories in the build phase?

**Comment by sikhote** [comment]
Seems related to https://github.com/zeit/now-examples/issues/150

**Comment by bukinoshita** [comment]
Is there a way to deploy NextJS + Express? I can't deploy any of my apps on v2

**Comment by paulogdm** [comment]
@bukinoshita It is not necessary to use the custom server with now v2.
With our routes configuration, everything can be done in that section.
Do you have a public repo that I could take a look and assist you with?

**Comment by bukinoshita** [comment]
@paulogdm https://github.com/infinity-co/infinity-site

**Comment by sikhote** [comment]
@paulogdm the method above may work, but I was unable to get v1 of now to work with paramterized routing. I had to use now v1 of now with docker, like here: https://github.com/sikhote/dougpalm.com/tree/78f29adcf063aa539d5e97fa03a868c7c3354826

**Comment by pedro** [comment]
Same issue here (I don't have access to v1).

It doesn't seem like `@now/node-server` runs the `now-build` script. Same for `@now/node`, from what I can tell neither runs that script, so I can't figure out a way to trigger it. Tried using `postinstall`, but that seems to run in a diff context.

**Comment by paulogdm** [comment]
Please, take a look at https://zeit.co/guides/custom-next-js-server-to-routes/

**Comment by sikhote** [comment]
@paulogdm thanks for posting, this was the guide I think a lot of us needed.

**Comment by timothyis** [comment]
Due to there being a guide for this issue now, I'm going to go ahead and close it! There is, however, another issue that calls for an example for this #229 which we will use to track progress on an example 🙏
